client/signup/main.jsx

```diff
@@ -785,9 +785,12 @@ class Signup extends Component {
 			};
 		}
 
+		let stepName = getStepModuleName( this.props.stepName );
+		stepName = stepName === 'user' ? 'user' : stepName;
+
 		return (
 			<div className="signup__step" key={ stepKey }>
-				<div className={ `signup__step is-${ kebabCase( this.props.stepName ) }` }>
+				<div className={ `signup__step is-${ stepName }` }>
```

Comment:
We're displaying the vertically-split login page if the `.is-user` class matches. But the thing is the step name must match `user`, otherwise, it's not applied. Since the `user-hosting` step is using the `user` component, we must extract it from the step name and pass it to the class name to style the step properly.

Reply:
Hi @zaguiini! This change caused some styling issues with a few signup steps, as there are CSS styles with selectors like `signup__step.is-<step name>`.
https://github.com/search?q=repo%3AAutomattic%2Fwp-calypso%20.signup__step.is&type=code

One example is the page picker step in the BBE flow:
| Before | After |
|--------|--------|
|<img width="1919" alt="image" src="https://user-images.githubusercontent.com/5436027/231353790-a4d6c2fa-bca2-4293-ac1b-0afdcda40c5c.png">|<img width="1913" alt="image" src="https://user-images.githubusercontent.com/5436027/231353872-72aa1287-77fa-4941-a66c-21510d457d20.png"> |

What do you think is the best step forward? How about something like this in L789?
```
stepName = stepName === 'user' ? 'user' : this.props.stepName;
```

I think having this distinction between step name and step module name is important because two different steps might have the same step module, and need to have different styles.

Reply:
@aneeshd16 It looks like we're the ones who wanted to do something special for the `user-hosting` step here, so rather than changing the behaviour for every flow in the signup framework we should have only made the change for our new step.

I'm thinking of something more like
```js
const stepName = this.props.stepName === 'user-hosting' ? 'user' : this.props.stepName
```

The `.is-xyz` classes have always been used for styling steps, not components. If there are some CSS rules that would be more appropriately tied to the React component so they get shared by everyone who uses the component regardless of what step it is (which I think is what @zaguiini was thinking) then it'd be better to migrate those CSS rules to the component's stylesheet.
To me, using `getStepModuleName()` is sort of like trying to associate something with the React component.

Reply:
Yeah, that was exactly my intention. Sorry for the regression -- let's fix it.

Reply:
@aneeshd16 here you go: https://github.com/Automattic/wp-calypso/pull/75611